### PR TITLE
refactor: replaced setState with riverpod state manager

### LINKED ISSUE #9: Code refactoring
Using Riverpod instead of setState as a state manager.

### PR DESCRIPTION
**Pull request by anyarylova**
#9  Issue solved

### CODE REVIEW

**Comment by anyarylova** [comment]
Tests should fail because they need to be rewritten to fit the new project structure

File: lib/main.dart [hunk 1]
```diff
@@ -1,11 +1,12 @@
 import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
```
Comment (cuprum-acid) [inline lib/main.dart:2]: Using Flutter's riverpod is issential for efficient and scalable code!

File: pubspec.yaml [hunk 2]
```diff
@@ -37,9 +37,9 @@ dependencies:
   cupertino_icons: ^1.0.8
   shared_preferences: ^2.3.2
   dio: ^5.6.0
+  flutter_riverpod: ^2.5.1
```
Comment (cuprum-acid) [inline pubspec.yaml:40]: 👍😊

File: lib/providers/situation_provider.dart
Comment (cuprum-acid) [inline lib/providers/situation_provider.dart:1]: Nice provider, but need to be localized!

**Review by cuprum-acid — APPROVED** [review]
Overall, it's a good idea to structure a project like this, but new tests should be introduced

**Pull request merged**